definitions/npm/chalk_v2.x.x/flow_v0.25.x-/chalk_v2.x.x.js

```diff
@@ -1,105 +1,90 @@
+  bgHsv(h: number, s: number, v: number): Chalk,
+  bgHwb(h: number, w: number, b: number): Chalk,
+  hex(color: string): Chalk,
+  keyword(color: string): Chalk,
+
+  +reset: Chalk,
+  +bold: Chalk,
+  +dim: Chalk,
+  +italic: Chalk,
+  +underline: Chalk,
+  +inverse: Chalk,
+  +hidden: Chalk,
+  +strikethrough: Chalk,
 
-  // General
-  declare var reset: $npm$chalk$Chain;
-  declare var bold: $npm$chalk$Chain;
-  declare var dim: $npm$chalk$Chain;
-  declare var italic: $npm$chalk$Chain;
-  declare var underline: $npm$chalk$Chain;
-  declare var inverse: $npm$chalk$Chain;
-  declare var strikethrough: $npm$chalk$Chain;
+  +visible: Chalk,
 
-  // Text colors
-  declare var black: $npm$chalk$Chain;
-  declare var red: $npm$chalk$Chain;
-  declare var redBright: $npm$chalk$Chain;
-  declare var green: $npm$chalk$Chain;
-  declare var greenBright: $npm$chalk$Chain;
-  declare var yellow: $npm$chalk$Chain;
-  declare var yellowBright: $npm$chalk$Chain;
-  declare var blue: $npm$chalk$Chain;
-  declare var blueBright: $npm$chalk$Chain;
-  declare var magenta: $npm$chalk$Chain;
-  declare var magentaBright: $npm$chalk$Chain;
-  declare var cyan: $npm$chalk$Chain;
-  declare var cyanBright: $npm$chalk$Chain;
-  declare var white: $npm$chalk$Chain;
-  declare var whiteBright: $npm$chalk$Chain;
-  declare var gray: $npm$chalk$Chain;
-  declare var grey: $npm$chalk$Chain;
+  +black: Chalk,
+  +red: Chalk,
+  +green: Chalk,
+  +yellow: Chalk,
+  +blue: Chalk,
+  +magenta: Chalk,
+  +cyan: Chalk,
+  +white: Chalk,
+  +gray: Chalk,
+  +grey: Chalk,
+  +blackBright: Chalk,
+  +redBright: Chalk,
+  +greenBright: Chalk,
+  +yellowBright: Chalk,
+  +blueBright: Chalk,
+  +magentaBright: Chalk,
+  +cyanBright: Chalk,
+  +whiteBright: Chalk,
 
-  // Background colors
-  declare var bgBlack: $npm$chalk$Chain;
-  declare var bgBlackBright: $npm$chalk$Chain;
-  declare var bgRed: $npm$chalk$Chain;
-  declare var bgRedBright: $npm$chalk$Chain;
-  declare var bgGreen: $npm$chalk$Chain;
-  declare var bgGreenBright: $npm$chalk$Chain;
-  declare var bgYellow: $npm$chalk$Chain;
-  declare var bgYellowBright: $npm$chalk$Chain;
-  declare var bgBlue: $npm$chalk$Chain;
-  declare var bgBlueBright: $npm$chalk$Chain;
-  declare var bgMagenta: $npm$chalk$Chain;
-  declare var bgMagentaBright: $npm$chalk$Chain;
-  declare var bgCyan: $npm$chalk$Chain;
-  declare var bgCyanBright: $npm$chalk$Chain;
-  declare var bgWhite: $npm$chalk$Chain;
-  declare var bgWhiteBright: $npm$chalk$Chain;
+  +bgBlack: Chalk,
+  +bgRed: Chalk,
+  +bgGreen: Chalk,
+  +bgYellow: Chalk,
+  +bgBlue: Chalk,
+  +bgMagenta: Chalk,
+  +bgCyan: Chalk,
+  +bgWhite: Chalk,
+  +bgBlackBright: Chalk,
+  +bgRedBright: Chalk,
+  +bgGreenBright: Chalk,
+  +bgYellowBright: Chalk,
+  +bgBlueBright: Chalk,
+  +bgMagentaBright: Chalk,
+  +bgCyanBright: Chalk,
+  +bgWhiteBrigh: Chalk,
+
+  supportsColor: ColorSupport
+};
+
+declare module "chalk" {
+  declare module.exports: Chalk
```

Comment:
It seems to me like what you want is to have both named exports and a default export, in which case you could accomplish this with using ES6 exports.

Reply:
Should change anything?

I try:
```js
const chalk = require('chalk')`;
import { bold } from 'chalk';
```

This still work. Maybe you can give me a simple example?
I can know how to modify.

Reply:
In my opinion what you have is correct, I was just explaining the reason why you need to mix `module.exports` and `export`.

Reply:
Actually scratch that. Right now, all your types and interfaces are outside of the `declare module`. This will make them global, which is not what we want. You will want to move all the types to inside the `declare module`. For example:

```ts
declare module 'chalk' {
  declare type TemplateStringsArray = $ReadOnlyArray<string>;

  declare export type Level = $Values<{
    None: 0,
    Basic: 1,
    Ansi256: 2,
    TrueColor: 3
  }>;

  ...

  declare module.exports: Chalk;
}
```

Reply:
OK, I know.
I did not have enough experience on `flow`.
I will try to modify.

Thank you!